### PR TITLE
Document submission

### PR DESCRIPTION
**Pull request by TehMillhouse**
TODO: Some explanatory text on the document submission page. Also the submit button should probably be disabled while waiting and after the first success to prevent accidental double submissions.

### CODE REVIEW

File: js/typeaheadsource.js [hunk 1]
```diff
@@ -3,8 +3,12 @@ import sortBy from "lodash/collection/sortBy";
 import uniq from "lodash/array/uniq";
 
 /** Returns a function usable as a typeahead source filtering the given items 'name' property */
```
Comment (Kha) [inline js/typeaheadsource.js]: update docstring

File: js/typeaheadsource.js [hunk 2]
```diff
@@ -3,8 +3,12 @@ import sortBy from "lodash/collection/sortBy";
...
-export default function makeSource(items) {
-  let sorted = sortBy(items, 'name');
+export default function makeSource(items, accessor) {
+  let sorted = [];
+  if (accessor)
+    sorted = sortBy(items, 'name');
```
Comment (TehMillhouse) [inline js/typeaheadsource.js]: TODO: make accessor string property

File: js/vm/documentsubmission.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+  setSubject(subject) {
+    this.subject = subject;
+  }
+
+  submit() {
+    /* input validation */
+    if (this.selectedLectures.length === 0) {
+      this.status = 'error';
+      this.errorlabel = "Bitte gib' die in der Prüfung geprüften Vorlesungen an.";
```
Comment (Kha) [inline js/vm/documentsubmission.js]: "gib"

File: js/vm/documentsubmission.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      return;
+    }
+    if (this.selectedExaminants.length === 0) {
+      this.status = 'error';
+      this.errorlabel = "Bitte gib' die in der Prüfung anwesenden PrüferInnen an.";
+      return;
+    }
+    if (!this.file || !this.file.name.toLowerCase().endsWith('.pdf')) {
+      this.status = 'error';
+      this.errorlabel = "Bitte gib' eine Pdf-Datei zum hochladen an.";
```
Comment (Kha) [inline js/vm/documentsubmission.js]: "Hochladen"

File: views/documentsubmission.html [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            items: ko.getObservable($data, 'selectedLectures'),
+            freeInput: true,
+            typeaheadjs: typeaheadDataset('lectures'),
+          }"></select>
+    </div>
+  </div>
+  <div class="form-group">
+    <label class="control-label col-md-3">An welcher Fakultät wird die Prüfung angeboten?&nbsp;</label>
+    <div class="col-md-8 btn-group btn-group-nowrap">
+      <button class="btn btn-default" data-bind="click: setSubject.bind($data, 'computer science'), css: { 'active': subject === 'computer science'}">Informatik</button>
```
Comment (Kha) [inline views/documentsubmission.html]: I'd prefer `click: function() { subject = 'computer science'; }`

**Pull request merged**